application/src/main/java/org/thingsboard/server/controller/plugin/TbWebSocketHandler.java

```diff
@@ -215,14 +206,14 @@ private TelemetryWebSocketSessionRef toRef(WebSocketSession session) throws IOEx
             this.asyncRemote = nativeSession.getAsyncRemote();
             this.sessionRef = sessionRef;
             this.msgQueue = new LinkedBlockingQueue<>(maxMsgQueuePerSession);
-            this.lastActivityTime = new AtomicLong(System.currentTimeMillis());
+            this.lastActivityTime = System.currentTimeMillis();
         }
 
         synchronized void sendPing() {
             try {
-                if (System.currentTimeMillis() - lastActivityTime.get() >= pingTimeout) {
+                if (System.currentTimeMillis() - lastActivityTime >= pingTimeout) {
```

Comment:
please extract System.currentTimeMillis() to a variable and call it only once per all sessions and not twice per session.